src/libasr/pass/intrinsic_functions.h

```diff
@@ -19,6 +19,7 @@ the code size.
 
 enum class IntrinsicElementalFunctions : int64_t {
     ObjectType,
+    Reversed,
```

Comment:
`Reversed` returns an iterator, but does not actually `reverse` a list. I think we do not have support for iterators yet in LPython.

The approach in this PR reverses the entire list, which I think is not the correct approach.